src/app/search/search.component.ts

```diff
@@ -443,14 +396,254 @@ export class SearchComponent implements OnInit {
+    if (this.sortModeMap.has(key)) {
+      switch (this.sortModeMap.get(key).SortBy) {
+        case true: {
+          order = {
+            _count: this.sortModeMap.get(key).CountOrderBy ? 'asc' : 'desc'
+          };
+          break;
+        }
+        case false: {
+          order = {
+            _term: this.sortModeMap.get(key).AlphabetOrderBy ? 'asc' : 'desc'
+          };
+          break;
+        }
+        default: {
+          order = {
+            _count: 'desc'
+          };
+          break;
+        }
+      }
+    } else {
+      order = {
+        _count: 'desc'
+      };
+    }
+    return order;
+  }
+  /**
+   * This handles selection of one filter, either taking it out from the list of active filters
+   * or adding it if not present
+   * @param category
+   * @param categoryValue
+   */
+  handleFilters(category: string, categoryValue: string) {
+    if (this.filters.has(category) && this.filters.get(category).has(categoryValue)) {
+      this.filters.get(category).delete(categoryValue);
+      // wipe out the category if empty
+      if (this.filters.get(category).size === 0) {
+        this.filters.delete(category);
+      }
+    } else {
+      if (!this.filters.has(category)) {
+        this.filters.set(category, new Set<string>());
+      }
+      this.filters.get(category).add(categoryValue);
+    }
+  }
+  sortByAlphabet(orderedArray, orderMode): any {
+    orderedArray = orderedArray.sort(function (a, b) {
+      if (orderMode) {
+        return a.key > b.key ? 1 : -1;
+      } else  {
+        return a.key < b.key ? 1 : -1;
+      }
+    });
+    return orderedArray;
+  }
+
+  sortByCount(orderedArray, orderMode): any {
+    orderedArray = orderedArray.sort(function (a, b) {
+      if (a.value < b.value) {
+        return !orderMode ? 1 : -1;
+      } else if (a.value === b.value) {
+        return a.key > b.key ? 1 : -1;
+      } else {
+        return !orderMode ? -1 : 1;
+      }
+    });
+    return orderedArray;
+  }
+
+  sortCategoryValue (valueMap: any, sortMode: boolean, orderMode: boolean): any {
+    let orderedArray = <any>[];
+    valueMap.forEach(
+      (value, key) => {
+        orderedArray.push(
+          {
+            key: key,
+            value: value
+          });
+      });
+    if (!sortMode) {
+      orderedArray = this.sortByAlphabet(orderedArray, orderMode);
+    } else {
+      orderedArray = this.sortByCount(orderedArray, orderMode);
+    };
+    const tempMap: Map<string, string> = new Map<string, string>();
+    orderedArray.forEach(
+      entry => {
+        tempMap.set(entry.key, entry.value);
+      });
+    return tempMap;
+  }
+
+  setTabActive() {
+    if (this.toolHits.length === 0 && this.workflowHits.length > 0) {
+      this.activeToolBar = false;
+    } else if (this.workflowHits.length === 0 && this.toolHits.length > 0) {
+      this.activeToolBar = true;
+    } else {
+      this.activeToolBar = true;
+    }
+  }
+}
+
+export class CategorySort {
```

Comment:
Rule of One.  One class per file, probably move these 2 to the model folder or a new model folder